R/xml_serialize.R

```diff
@@ -0,0 +1,49 @@
+#' Serializing XML objects to connections.
+#'
+#' @inheritParams base::serialize
+#' @param ... Additional arguments passed to \code{\link{read_xml}}.
+#' @examples
+#' library(xml2)
+#' x <- read_xml("<a>
+#'   <b><c>123</c></b>
+#'   <b><c>456</c></b>
+#' </a>")
+#'
+#' b <- xml_find_all(x, "//b")
+#' out <- xml_serialize(b, NULL)
+#' xml_unserialize(out)
+#' @export
+xml_serialize <- function(object, connection, ...) UseMethod("xml_serialize")
+
+#' @export
+xml_serialize.xml_document <- function(object, connection, ...) {
+  serialize(structure(as.character(object, ...), class = "xml_document"), connection)
```

Comment:
These should probably use unique class names like `xml_serialised_document` (just so if you load it another way you don't get confused)